platform/linux-dpdk/odp_crypto.c

```diff
@@ -116,13 +124,14 @@ static int auth_alg_odp_to_rte(odp_auth_alg_t auth_alg,
 	switch (auth_alg) {
 	case ODP_AUTH_ALG_NULL:
 		auth_xform->auth.algo = RTE_CRYPTO_AUTH_NULL;
+		auth_xform->auth.digest_length = 0;
 		break;
 	case ODP_AUTH_ALG_MD5_HMAC:
 #if ODP_DEPRECATED_API
 	case ODP_AUTH_ALG_MD5_96:
 #endif
 		auth_xform->auth.algo = RTE_CRYPTO_AUTH_MD5_HMAC;
-		auth_xform->auth.digest_length = 12;
+		auth_xform->auth.digest_length = 16;
```

Comment:
This is wrong. MD5_96 should use 12-byte digest length. MD5_HMAC should use digest length specified by application.

Reply:
openssl PMD in 17.02 supports the digest length to be min 16 bytes. But I have now removed the setting digest_lengths from here as it sets it from the application.

Reply:
For deprecated MD5_96 (and SHA256_128/AES128_GCM) application won't set `auth.digest_len`, so you have to enforce 12 and 16 bytes digest len. For MD5_HMAC/SHA256_HMAC (and others) application will set `auth.digest_len`.

IIRC QAT driver should support truncated digests out-of-box. Also you can try expanding pmd_openssl to support truncated lengths, it should not be very complicated. Also note, that full-length HMAC won't be useful for IPsec.

Reply:
setting digest_length is moved to get_crypto_dev(), where it reads from PMD and set the min digest length if application doesn't set it. Reagrding the QAT, I need to look at it and understand. may be as a separate PR.

Reply:
Implementation should not set digest len for cases other than several deprecated auth algos.